storagenode/gracefulexit/worker.go

```diff
@@ -169,6 +122,78 @@ func (worker *Worker) Run(ctx context.Context, done func()) (err error) {
 	return errs.Wrap(err)
 }
 
+func (worker *Worker) transferPiece(ctx context.Context, transferPiece *pb.TransferPiece, c pb.SatelliteGracefulExit_ProcessClient) error {
+	pieceID := transferPiece.OriginalPieceId
+	reader, err := worker.store.Reader(ctx, worker.satelliteID, pieceID)
+	if err != nil {
+		transferErr := pb.TransferFailed_UNKNOWN
+		if errs.Is(err, os.ErrNotExist) {
+			transferErr = pb.TransferFailed_NOT_FOUND
+		}
+		worker.log.Error("failed to get piece reader.", zap.Stringer("satellite ID", worker.satelliteID), zap.Stringer("piece ID", pieceID), zap.Error(errs.Wrap(err)))
+		worker.handleFailure(ctx, transferErr, pieceID, c.Send)
+		return err
+	}
+
+	addrLimit := transferPiece.GetAddressedOrderLimit()
+	pk := transferPiece.PrivateKey
+
+	originalHash, originalOrderLimit, err := worker.getHashAndLimit(ctx, reader, addrLimit.GetLimit())
+	if err != nil {
+		worker.log.Error("failed to get piece hash and order limit.", zap.Stringer("satellite ID", worker.satelliteID), zap.Stringer("piece ID", pieceID), zap.Error(errs.Wrap(err)))
+		worker.handleFailure(ctx, pb.TransferFailed_UNKNOWN, pieceID, c.Send)
+		return err
+	}
+
+	putCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// TODO what's the typical expiration setting?
+	pieceHash, peerID, err := worker.ecclient.PutPiece(putCtx, ctx, addrLimit, pk, reader, time.Now().Add(time.Second*600))
+	if err != nil {
+		if piecestore.ErrVerifyUntrusted.Has(err) {
+			worker.log.Error("failed hash verification.", zap.Stringer("satellite ID", worker.satelliteID), zap.Stringer("piece ID", pieceID), zap.Error(errs.Wrap(err)))
+			worker.handleFailure(ctx, pb.TransferFailed_HASH_VERIFICATION, pieceID, c.Send)
+		} else {
+			worker.log.Error("failed to put piece.", zap.Stringer("satellite ID", worker.satelliteID), zap.Stringer("piece ID", pieceID), zap.Error(errs.Wrap(err)))
+			// TODO look at error type to decide on the transfer error
+			worker.handleFailure(ctx, pb.TransferFailed_STORAGE_NODE_UNAVAILABLE, pieceID, c.Send)
+		}
+		return err
```

Comment:
nit - these should maybe be wrapped to be consistent, since the `Error.New`s will have the graceful exit prefix

Reply:
Nothing is done with the error after it is returned (we just do `continue`) so I will not worry about this for now.